relay-server/src/actors/healthcheck.rs

```diff
@@ -141,10 +174,12 @@ impl Healthcheck {
                 let service = main_service.clone();
 
                 tokio::spawn(async move {
-                    let response = match message.data {
-                        HealthcheckMessage::Health(data) => service.handle_is_healthy(data).await,
+                    match message {
+                        HealthcheckEnvelope::IsHealthy(msg, response_tx) => {
+                            let response = service.handle_is_healthy(msg).await;
```

Comment:
We could now introduce something like `ServiceMessage::handle(self, service: &S)` which contains the async definition of the handler and returns the result. That would sadly require async_trait though. Also, I have no good answer to model the fire-and-forget case.

Reply:
I'm going to leave this as potentially future work.  I want to wait a bit longer until I see the patterns better.